app/models/qa/linked_data/config/context_property_map.rb

```diff
@@ -70,12 +95,41 @@ def extract_label
             default
           end
 
-          def ldpath_program
-            return @program if @program.present?
+          def basic_program
+            @basic_program ||= ldpath_program(ldpath)
+          end
+
+          def expansion_label_program
+            @expansion_label_program ||= ldpath_program(expansion_label_ldpath)
+          end
+
+          def expansion_id_program
+            @expansion_id_program ||= ldpath_program(expansion_id_ldpath)
+          end
+
+          def ldpath_program(ldpath)
             program_code = ""
             prefixes.each { |key, url| program_code << "@prefix #{key} : <#{url}> \;\n" }
             program_code << "property = #{ldpath} \;"
-            @program = Ldpath::Program.parse program_code
+            Ldpath::Program.parse program_code
+          end
+
+          def ldpath_evaluate(program, graph, subject_uri)
+            output = program.evaluate subject_uri, graph
+            output.present? ? output['property'].uniq : nil
+          rescue
+            'PARSE ERROR'
```

Comment:
@elrayle I know this is not new code, but where does the string show up to the user?  I'm wondering if it should be logged.

Reply:
@cam156 Are you asking about the 'PARSE ERROR' string?  This is a stop gap from the ldpath gem failing to parse some ldpaths.  So it would show up here...
```
{
    group: "Hierarchy",
    property: "Narrower",
    values: [ 'PARSE ERROR' ],
    selectable: true,
    drillable: true
},
```

Not ideal.  Better is perhaps logging and returning nothing for the property that wouldn't parse or `values: ['']`

Reply:
@elrayle I was just wondering if it was something people would notice.  Rescuing errors and silently handling them is something I have done and regretted.

Reply:
With 2nd commit, reworked the results so that the error is in a separate field in the hash...

```
{
    group: "Hierarchy",
    property: "Narrower",
    values: [],
    selectable: true,
    drillable: true,
    error: "LDPATH PARSE ERROR (See log for more information)"
},
```